include/direwolf/vulkan/vulkansetup.h

```diff
@@ -12,30 +47,22 @@
 namespace dw {
 namespace vulkan {
 
-class TempVulkanSetupObject {
-public:
-    // TODO: add struct to hold initialize data.. e.g. desiredExtensions and names etc. See createVulkanInstance implementation to get an idea of what it should hold
-    TempVulkanSetupObject(std::vector<const char*>* desiredExtensions = nullptr);
-    bool initialize(std::vector<const char*>* desiredExtensions);
-    bool isValid() const { return m_isValid; }
-    static std::vector<const char*> getDefaultInstanceExtensions();
-
-private: // variables
-    bool m_isValid;
-    VkInstance m_instance;
-
-private: // functions
-    bool initLibs();
-    bool initProcAddr();
-    bool isExtensionSupported(const char* extension, std::vector<VkExtensionProperties>* availableExtensions = nullptr) const;
-    bool loadGlobalLevelFunctions();
-    bool loadInstanceLevelFunctions();
-    bool getPhysicalDevices(std::vector<VkPhysicalDevice>& outAvailableDevices);
-    bool getPhysicalDeviceExtensions(const VkPhysicalDevice& device, std::vector<VkExtensionProperties>& outAvailableExtensions);
-    bool loadInstanceLevelFunctionsFromExtensions(const std::vector<const char*>* enabledExtensions = nullptr);
-    bool createVulkanInstance(std::vector<const char*>* desiredExtensions = nullptr);
-    bool getAvailableInstanceExtensions(std::vector<VkExtensionProperties>& outAvailableExtensions) const;
+struct VulkanInstanceInitData {
+    char* applicationName = "NullNameApplication";
+    char* engineName = "NullNameEngine";
+    std::vector<VkExtensionProperties>* availableExtensionsHandle = nullptr;
+    std::vector<const char*>* desiredExtensions = nullptr;
+    // TODO: Extend this with more options?
 };
 
+bool InitializeVulkan();
```

Comment:
Nice, function names now with UpperCase

Reply:
Yeah, I don't really like this.. but I was thinking, maybe public functions should be uppercase and private lowercase? It's just a thought

Reply:
Yeah maybe that can work 👍